src/libraries/System.IO.FileSystem/tests/Base/SymbolicLinks/BaseSymbolicLinks.FileSystem.cs

```diff
@@ -0,0 +1,348 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Diagnostics;
+using Xunit;
+
+namespace System.IO.Tests
+{
+    // Contains test methods that can be used for FileInfo, DirectoryInfo, File or Directory.
+    public abstract class BaseSymbolicLinks_FileSystem : BaseSymbolicLinks
+    {
+        /// <summary>Creates a new file or directory depending on the implementing class.
+        /// If createOpposite is true, creates a directory if the implementing class is for File or FileInfo, or
+        /// creates a file if the implementing class is for Directory or DirectoryInfo.</summary>
+        protected abstract void CreateFileOrDirectory(string path, bool createOpposite = false);
+        protected abstract void AssertIsCorrectTypeAndDirectoryAttribute(FileSystemInfo fsi);
+        protected abstract void AssertLinkExists(FileSystemInfo link);
+        protected abstract void AssertExistsWhenNoTarget(FileSystemInfo link);
+        /// <summary>Calls the actual public API for creating a symbolic link.</summary>
+        protected abstract FileSystemInfo CreateSymbolicLink(string path, string pathToTarget);
+        /// <summary>Calls the actual public API for resolving the symbolic link target.</summary>
+        protected abstract FileSystemInfo ResolveLinkTarget(string linkPath, string? expectedLinkTarget, bool returnFinalTarget = false);
+
+        /// <summary>
+        /// Appends \??\ to the absolute path
+        /// </summary>
+        protected void AssertPathEquals(string expected, string actual)
+        {
+#if WINDOWS // Windows implementation has a couple nuances compared to Unix.
+            // If the link target path is absolute, the following occurs:
+            // ResolveLinkTarget(), which uses DeviceIoControl, will return the path with the DOS Device prefix (\??\).
+            // ResolveLinkTarget(returnFinalTarget: true), which uses GetFinalPathNameByHandleW, will return with the extended prefix (\\?\).
+            // For testing, we will ignore this prefix.
```

Comment:
@JeremyKuhne @adamsitnik what are your thoughts on the differences of the DOS device prefixes described in this comment? 
* Would you suggest to keep path consistency and change one prefix to the other
* Or we should keep them as-is (regardless of the inconsistency) and let the user decide what to do with each one
* Or we should "normalize" and always remove the prefixes (even though the prefix indicates that the path is already normalized).

Reply:
I would never let `\??\` paths out. While there are a few APIs that tolerate it, many do not. All paths should be returned as Win32 paths (`\\?\`). Change `\??\` to `\\?\`.

You can't remove `\\?\` as you won't be able to access paths that require it (paths that have trailing spaces, paths that aren't rooted to a drive letter).

Reply:
Thanks.